sympy/printing/c.py

```diff
@@ -396,8 +397,11 @@ def _print_MatrixElement(self, expr):
 
     def _print_Symbol(self, expr):
         name = super()._print_Symbol(expr)
-        # if name is an indexed, e.g. a[0], only check the variable name
-        self._check_valid_c_variable_name(name.split('[')[0])
+        if self._settings['strict']:
+            # if name is an accessor, e.g. a[0], a.b a->b, only check the first
+            # variable name
+            first = name.split('[')[0].split('.')[0].split('->')[0]
```

Comment:
When applied to a struct, this references the struct name and not the variable name.

Reply:
Also will this disallow an array of structs? 
`structname[1].variable_name`

Hmm... how about an array of arrays?
`array[0][1]`

Reply:
my earlier comment applies here as well..